### PR TITLE
Hook actual durability for Pickonimbus/Drills

### PR DESCRIPTION
**Pull request by 0xallie**
> [!NOTE]  
> I recommend using "Squash and merge" when merging this pull request to avoid an unclean commit history.

Instead of the previous DrawContext hack.

Confirmed working with Durability Notifier on Fabric 1.20.2.

### CODE REVIEW

File: src/main/java/me/xmrvizzy/skyblocker/mixin/DrawContextMixin.java
Comment (AzureAaron) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/DrawContextMixin.java:1]: Where did my attribute shard display info feature go 😭😭😭

Comment (AzureAaron) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/DrawContextMixin.java:1]: Where did my attribute shard display info feature go 😭😭😭

**Review by AzureAaron — CHANGES_REQUESTED** [review]

File: src/main/java/me/xmrvizzy/skyblocker/mixin/DrawContextMixin.java
Comment (0xallie) [reply src/main/java/me/xmrvizzy/skyblocker/mixin/DrawContextMixin.java:1]: Oh let me see...

Comment (0xallie) [reply src/main/java/me/xmrvizzy/skyblocker/mixin/DrawContextMixin.java:1]: Re-added.

File: src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java [hunk 1]
```diff
@@ -15,6 +19,56 @@
...
+        }
+
+        NbtCompound extraAttributes = tag.getCompound("ExtraAttributes");
+        if (!extraAttributes.contains("drill_fuel") && !extraAttributes.getString("id").equals("PICKONIMBUS")) {
+            return null;
+        }
+
+        int current = 0;
+        int max = 0;
+        String clearFormatting = "";
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java:43]:
```suggestion
        String clearFormatting;
```

File: src/main/java/me/xmrvizzy/skyblocker/mixin/ItemStackMixin.java [hunk 2]
```diff
@@ -30,4 +32,31 @@ public abstract class ItemStackMixin {
 
 		return original;
 	}
+
+	@ModifyReturnValue(method = "getDamage", at = @At("RETURN"))
+	private int skyblocker$handlePickoDrillDamage(int original) {
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/ItemStackMixin.java:37]:
```suggestion
	private int skyblocker$handleDamage(int original) {
```

File: src/main/java/me/xmrvizzy/skyblocker/mixin/ItemStackMixin.java [hunk 3]
```diff
@@ -30,4 +32,31 @@ public abstract class ItemStackMixin {
...
+		Durability dur = ItemUtils.getDurability((ItemStack) (Object) this);
+		if (dur != null) {
+			return dur.max() - dur.current();
+		}
+		return original;
+	}
+
+	@ModifyReturnValue(method = "getMaxDamage", at = @At("RETURN"))
+	private int skyblocker$handlePickoDrillMaxDamage(int original) {
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/ItemStackMixin.java:46]:
```suggestion
	private int skyblocker$handleMaxDamage(int original) {
```

File: src/main/java/me/xmrvizzy/skyblocker/mixin/ItemStackMixin.java [hunk 4]
```diff
@@ -30,4 +32,31 @@ public abstract class ItemStackMixin {
...
+		Durability dur = ItemUtils.getDurability((ItemStack) (Object) this);
+		if (dur != null) {
+			return dur.max();
+		}
+		return original;
+	}
+
+	@ModifyReturnValue(method = "isDamageable", at = @At("RETURN"))
+	private boolean skyblocker$handlePickoDrillDamageable(boolean original) {
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/ItemStackMixin.java:55]:
```suggestion
	private boolean skyblocker$handleDamageable(boolean original) {
```

File: src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java [hunk 5]
```diff
@@ -15,6 +19,56 @@
 public class ItemUtils {
     private final static Pattern WHITESPACES = Pattern.compile("^\\s*$");
 
+    public record Durability(int current, int max) {
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java:22]: Can you move this to the end of the class please?

File: src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java [hunk 6]
```diff
@@ -15,6 +19,56 @@
...
+    }
+
+    @Nullable
+    public static Durability getDurability(ItemStack stack) {
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java:26]: Maybe move this to the end before `getSkyblockerStack`?

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

File: src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java (hunk 1 above)
Comment (0xallie) [reply src/main/java/me/xmrvizzy/skyblocker/utils/ItemUtils.java:43]: Yeah good point, I just copied the existing code but will do.

**Comment by 0xallie** [comment]
All suggestions should be applied now (I recommend squash merging to avoid unnecessary commits).

**Review by kevinthegreat1 — DISMISSED** [review]
Works great, thanks for contributing!

**Review by AzureAaron — DISMISSED** [review]

**Review by LifeIsAParadox — APPROVED** [review]

**Pull request merged**